fdbclient/include/fdbclient/TenantManagement.actor.h

```
@@ -369,6 +369,33 @@ Future<std::vector<std::pair<TenantName, TenantMapEntry>>> listTenants(Reference
 	}
 }
 
+ACTOR template <class Transaction>
+Future<Void> renameTenantTransaction(Transaction tr, TenantNameRef oldName, TenantNameRef newName) {
+	tr->setOption(FDBTransactionOptions::RAW_ACCESS);
+	state Optional<TenantMapEntry> oldEntry;
+	state Optional<TenantMapEntry> newEntry;
+	wait(store(oldEntry, tryGetTenantTransaction(tr, oldName)) &&
+	     store(newEntry, tryGetTenantTransaction(tr, newName)));
+	if (!oldEntry.present()) {
+		throw tenant_not_found();
```

This is semantically different from the create and delete transactions, which return successfully if the tenant does or does not exist, respectively.

This isn't a problem for the `fdbcli` version, since we know it is checking for the state of the tenants in advance, so we only need to worry about what happens with direct special key-space accesses.

The main issue with this is that if someone were to try to rename a tenant in a retry loop without checking the existence of the tenants, we may end up throwing a `tenant_not_found` or `tenant_already_exists` error even though the rename actually happened. The alternative is to silently return from this function without doing any rename in the event that a rename cannot happen because of missing or conflicting tenants, which is similar to what create and delete do. In this case, though, it's also misleading to report success if we didn't do the rename, because unlike the create/delete case the state of the cluster doesn't match what you tried to do.

I think, then, that I agree that your current implementation is fine. We should document this behavior, though, so users know to check for the tenant's existence first.